diff --git a/plugins/rewrite.inc b/plugins/rewrite.inc
index 31bb93f..a77c4f5 100644
--- a/plugins/rewrite.inc
+++ b/plugins/rewrite.inc
@@ -26,6 +26,12 @@ function feeds_tamper_rewrite_form($importer, $element_key, $settings) {
     '#title' => t('Replacement pattern'),
     '#default_value' => isset($settings['text']) ? $settings['text'] : '',
   );
+  $form['only_if_empty'] = array(
+    '#type' => 'checkbox',
+    '#title' => t('Only if empty'),
+    '#default_value' => isset($settings['only_if_empty']) ? $settings['only_if_empty'] : 0,
+    '#description' => t('This field will be set to the value specified only if the imported field is empty.'),
+  );
   $form['help'] = array(
     '#type' => 'fieldset',
     '#title' => t('Available Replacement Patterns'),
@@ -37,10 +43,12 @@ function feeds_tamper_rewrite_form($importer, $element_key, $settings) {
 }
 
 function feeds_tamper_rewrite_callback($result, $item_key, $element_key, &$field, $settings, $source) {
-  $trans = array();
-  $item = $result->items[$item_key];
-  foreach ($item as $key => $value) {
-    $trans['[' . $key . ']'] = is_array($value) ? reset($value) : $value;
+  if ((!empty($settings['only_if_empty']) && !$field) || empty($settings['only_if_empty'])) {
+    $trans = array();
+    $item = $result->items[$item_key];
+    foreach ($item as $key => $value) {
+      $trans['[' . $key . ']'] = is_array($value) ? reset($value) : $value;
+    }
+    $field = strtr($settings['text'], $trans);
   }
-  $field = strtr($settings['text'], $trans);
 }
